js/game.js

```diff
@@ -164,4 +196,96 @@ Game.prototype.warning = function (color) {
   return result;
 };
 
+Game.prototype.pieceToAN = function(x, y) {
+  var xChars = 'abcdefgh';
+  return xChars.charAt(x) + (8 - y);
+}
+
+Game.prototype.gameToFEN = function () {
+  var board = this.board;
+  var turn = this.turn;
+  var FEN = '';
+ 
+  // Convert the board configuration into FEN
+  var FENboard = '';
+  var missingPieces = 0;
+  for (var i = 0; i < 8; i++) {
+    for (var j = 0; j < 8; j++) {
+      if(board[i][j]) {
+        if (missingPieces) FENboard += missingPieces;
+        missingPieces = 0;
+        FENboard += board[i][j].FENname;
+      } else {
+        missingPieces++;
+      }
+    }
+    if (missingPieces) FENboard += missingPieces;
+    missingPieces = 0;    
+    FENboard += i < 7 ? '/' : '';
+  }
+ 
+  FEN += FENboard;
+ 
+  // Find the active colour
+  var activeColor =
+    turn.length &&
+    turn[turn.length - 1].color === 'W' ? 'b' : 'w';
+ 
+  FEN += ' ' + activeColor;
+ 
+  // Check castling availability
+  var possibleCastling = '';
+ 
+  var whiteKingMoved;
+  var blackKingMoved;
+  turn.some(function (turn) {

```

Comment:
It's probably better to use `forEach` instead of `some` here.

Reply:
Oh yeah, I'll replace that. Thanks.